setuptools_rust/build.py

```diff
@@ -349,10 +351,42 @@ def get_dylib_ext_path(self, ext, target_fname):
         assert modpath not in build_ext.ext_map
         build_ext.ext_map[modpath] = ext
         try:
-            return build_ext.get_ext_fullpath(target_fname)
+            ext_path = build_ext.get_ext_fullpath(target_fname)
         finally:
             del build_ext.ext_map[modpath]
 
+        if '.abi3.' in ext_path:
+            return ext_path
+        # Examples: linux_x86_64, linux_i686, manylinux2014_aarch64, manylinux_2_24_armv7l
+        plat_name = self.plat_name.lower().replace('-', '_').replace('.', '_')
+        if not plat_name.startswith(('linux', 'manylinux')):
+            return ext_path
+
+        arch_parts = []
+        arch_found = False
+        for item in plat_name.split('_'):
+            if item.startswith(('linux', 'manylinux')):
+                continue
+            if item.isdigit() and not arch_found:
+                # manylinux_2_24_armv7l arch should be armv7l
+                continue
+            arch_found = True
+            arch_parts.append(item)
+        target_arch = '_'.join(arch_parts)
+        host_platform = sysconfig.get_platform()
+        host_arch = host_platform.rsplit('-', 1)[1]
+        # Remove incorrect platform tag if we are cross compiling
+        if target_arch and host_arch != target_arch:
```

Comment:
Yikes. Do we also need to check abi here? e.g. `windows-gnu` versus `windows-msvc`?

Is it enough to check if `plat_name` doesn't match `sysconfig.get_platform()`?

Maybe we should remove the platform tag always, rather than try to do some clever detection? Depends if there's a reason to keep the platform tag around.

Reply:
Or alternatively, looking at discussion in #138 it sounds like Python also gets this wrong? Maybe we should not fix the platform tag, and instead just leave it to users. If that's consistent with what C does, it seems most reliable (even if it's annoying).

Reply:
> Maybe we should remove the platform tag always, rather than try to do some clever detection?

I don't know, but I did encounter an error once with PyPy with no platform tag, see https://github.com/PyO3/setuptools-rust/runs/2208718989?check_suite_focus=true

Reply:
> Or alternatively, looking at discussion in #138 it sounds like Python also gets this wrong? 

Yes, it's terrible.

> Maybe we should not fix the platform tag, and instead just leave it to users. If that's consistent with what C does, it seems most reliable (even if it's annoying).

It would make supporting `cross` kinda useless IMO.